### PR TITLE
Consider a product valid if missing an expiration date

### LINKED ISSUE #359: Bug in sample barcode scanning - sample is showing as expired.
The sample barcode scanned is coming up expired.  "Sample is expired!" "Please collect another sample."

This shouldn't be the case since there is no date assigned in the barcode. You are only collecting the assigned barcode ID

### LINKED ISSUE #448: App unexpected closes/crashes when saving sample shipment form
The application unexpectedly closes/crashes when saving the completed sample shipment form.

The test was run on a single phone 3 times.

### PR DESCRIPTION
**Pull request by vincent-karuri**
This closes #359 and closes #448.

### CODE REVIEW

File: rdt-open-reader/src/main/java/io/ona/rdt/widget/RDTBarcodeFactory.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ protected void moveToNextStep() {
     }
 
     private boolean isRDTExpired(Date date) {
-        return date == null ? true : new Date().after(date);
+        return date != null && new Date().after(date);
```
Comment (pld) [inline rdt-open-reader/src/main/java/io/ona/rdt/widget/RDTBarcodeFactory.java:89]: can we add a test for this?

Comment (vincent-karuri) [reply rdt-open-reader/src/main/java/io/ona/rdt/widget/RDTBarcodeFactory.java:89]: That is covered [here](https://github.com/onaio/rdt-standard/blob/dcb9b4c804c2a13d5534efda9334d1f0444691c0/rdt-open-reader/src/test/java/io/ona/rdt/widget/RDTBarcodeFactoryTest.java#L110) and [here](https://github.com/onaio/rdt-standard/blob/dcb9b4c804c2a13d5534efda9334d1f0444691c0/rdt-open-reader/src/test/java/io/ona/rdt/widget/RDTBarcodeFactoryTest.java#L116).

Comment (pld) [reply rdt-open-reader/src/main/java/io/ona/rdt/widget/RDTBarcodeFactory.java:89]: can you please add a test that fails with the old code and passes with the new code

Comment (vincent-karuri) [reply rdt-open-reader/src/main/java/io/ona/rdt/widget/RDTBarcodeFactory.java:89]: Done.

**Pull request merged**